source/D2Common/src/Drlg/DrlgDrlg.cpp

```diff
@@ -545,47 +545,29 @@ int* __fastcall DRLG_GetRoomCenterX_RoomWarpXFromRoom(D2RoomExStrc* pRoomEx)
 //D2Common.0x6FD74B40
 void __fastcall DRLG_ComputeLevelWarpInfo(D2DrlgLevelStrc* pLevel)
 {
-	int* pX = NULL;
-	int* pY = NULL;
-	int nCounter = 0;
-	int nFlags = 0;
-	BOOL bHasWaypoint = FALSE;
-
 	for (D2RoomExStrc* pRoomEx = pLevel->pFirstRoomEx; pRoomEx; pRoomEx = pRoomEx->pRoomExNext)
 	{
-		if (pRoomEx->dwFlags & (ROOMEXFLAG_HAS_WAYPOINT | ROOMEXFLAG_HAS_WAYPOINT_SMALL))
-		{
-			bHasWaypoint = TRUE;
-		}
+		// First check if we have a waypoint
+		bool bHasWarp = (pRoomEx->dwFlags & ROOMEXFLAG_HAS_WAYPOINT_MASK);
```

Comment:
It's not a big issue, but maybe it's better to explicitly return a `bool` from this check instead of relying on implicit conversion from `int` to `bool`

`bool bHasWarp = (pRoomEx->dwFlags & ROOMEXFLAG_HAS_WAYPOINT_MASK) != 0;`